src/data/list/nodup.lean

```diff
@@ -322,6 +322,26 @@ begin
   { simp [ne.symm H, H, update_nth, ← apply_ite (cons (f hd))] }
 end
 
+lemma nodup.pairwise_of_forall_ne {l : list α} {r : α → α → Prop}
+  (hl : l.nodup) (h : ∀ (a ∈ l) (b ∈ l), a ≠ b → r a b) : l.pairwise r :=
```

Comment:
It might be nice to also have this property when `hl` is replaced by `reflexive r`. And then we maybe we want their common generalization "`r` is reflexive on the elements that are duplicated in `l`". Do we have a nice way to define the duplicates of `l`? We could write `\forall x, count x l > 1 -> r x x`, but that requires `decidable_eq`...